File: src/interfaces/movements.ts
export interface InmateOption {
  id: string;
  label: string;
  cellule_id: string;
}

export interface CelluleOption {
  id: string;
  label: string;
}

export interface Release {
  id: string;
  inmate_id: string;
  inmate_name: string;
  release_date: string;
  reason: string;
  notes?: string | null;
  created_at: string;
  updated_at: string;
}

export interface Transfer {
  id: string;
  inmate_id: string;
  inmate_name: string;
  from_cellule_id: string;
  from_cellule_name: string;
  to_cellule_id: string;
  to_cellule_name: string;
  transfer_date: string;
  reason: string;
  notes?: string | null;
  created_at: string;
  updated_at: string;
}

File: src-tauri/src/models/movements.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct InmateOption {
    pub id: String,
    pub label: String,
    pub cellule_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct CelluleOption {
    pub id: String,
    pub label: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReleaseInput {
    pub inmate_id: String,
    pub release_date: String,
    pub reason: String,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Release {
    pub id: String,
    pub inmate_id: String,
    pub inmate_name: String,
    pub release_date: String,
    pub reason: String,
    pub notes: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TransferInput {
    pub inmate_id: String,
    pub to_cellule_id: String,
    pub transfer_date: String,
    pub reason: String,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Transfer {
    pub id: String,
    pub inmate_id: String,
    pub inmate_name: String,
    pub from_cellule_id: String,
    pub from_cellule_name: String,
    pub to_cellule_id: String,
    pub to_cellule_name: String,
    pub transfer_date: String,
    pub reason: String,
    pub notes: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}
